crates/wasmtime/src/module.rs

```diff
@@ -332,8 +336,28 @@ impl Module {
         }
     }
 
-    /// Compiles a binary-encoded WebAssembly module to an artifact usable by
-    /// Wasmtime.
+    /// Creates a new WebAssembly `Module` from the contents of the given
+    /// `file` on disk, but with assumptions that the file is 'sane'.
+    /// In other words, the file should be a binary- or text-format WebAssembly
+    /// module or a precompiled version of it generated by wasmtime.
+    ///
+    /// # Unsafety
+    ///
+    /// This function is marked as `unsafe` as it relies on [`Module::deserialize`].
+    #[cfg(compiler)]
+    #[cfg_attr(nightlydoc, doc(cfg(feature = "cranelift")))] // see build.rs
+    pub unsafe fn from_trusted_file(engine: &Engine, file: impl AsRef<Path>) -> Result<Module> {
+        let mmap = MmapVec::from_file(file.as_ref())?;
+        if &mmap[0..4] == b"\x7fELF" {
+            return SerializedModule::from_mmap(mmap, &engine.config().module_version)?
+                .into_module(engine);
```

Comment:
Unfortunately, things changed underneath you (in #5153), so `SerializedModule` doesn't exist any more. Fortunately, I think this is an easy fix. In `crates/wasmtime/src/engine.rs`, make `fn load_code` be `pub(crate)`. Then here you can call `engine.load_code(mmap, ObjectKind::Module)`. See `deserialize` and `deserialize_file` for examples of the current pattern.

Reply:
Oops, I forgot to `cargo check` again after rebasing. I'll fix eod today

Reply:
This is done and I found a small typo near load_code so fixed it too ;)